docs/pages/blog/2023-toolpad-beta-announcement.md

```diff
@@ -0,0 +1,81 @@
+
+Internal tools are software applications that are developed and used within an organization itself. They can automate tasks, manage data, and help foster collaboration. Internal tools are a valuable asset for businesses of all sizes, as they help to improve efficiency, productivity, and communication.
+
+A sub-optimal tool means a waste of time and effort for everyone. Maintaining them distracts developers from solving actual business problems. We've all had our fair share of wrestling with lousy internal tools that we hoped we'd never have to use more than once. In recent years, customer-facing software has moved quickly and is now flashy and sleek. It's time for internal tools to get this same upgrade.
+
+## What is Toolpad exactly?
+
+Toolpad is an open-source, low-code, drag-and-drop admin builder. The primary purpose of Toolpad is to create data-intensive React apps faster. Toolpad simplifies writing back-end logic, connecting to a data source, querying and mutating data, and deploying an app. It can quickly convert an API, script, or SQL query into a web UI.
+
+Toolpad is not ideal for building static web pages, mobile apps, or customer-facing custom-designed front ends. It's for building admin applications, CRUD interfaces, custom internal tools, and analytics dashboards.
+
+<img alt="Building an application on Toolpad " src="/static/blog/2023-toolpad-beta-announcement/toolpad.png" style="margin-bottom: 16px;" loading="lazy" width="2048" height="1536" />
+
+<iframe style="width:100%; height:350px; border:0; border-radius: 10px; overflow:hidden; margin-bottom: 24px"
+src="https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/master/examples/npm-stats?embed=1"
+title="Demo of a Toolpad app"
+></iframe>
+
+## Who is Toolpad for?
+
+Before the arrival of Toolpad, MUI primarily catered to the front-end needs of the development process. Our products were mostly useful for React engineers or designers. We saw many full-stack engineers use our advanced components for data-intensive apps. These highly customizable components are well-suited for complex apps, but not all apps require the same level of customization: for example, internal tools just need basic theming.
+
+Toolpad is for full-stack and back-end developers. They build, manage, and integrate data pipelines, server-side logic, databases, microservices, and APIs—and know the most about them. Sharing these endpoints and maintaining the internal documentation with the front-end teams is a hassle that lasts forever. Toolpad empowers you to take the next step and build the UI you need using our drag-and-drop editor.
+
+## What are the main features available today?
+
+### Use MUI Core and MUI X Pro components
+
+MUI Core provides all modern components; a chosen few are currently available inside Toolpad, and we're adding more all the time. MUI X Pro (Data Grid and Date Picker) components are also available for free in Toolpad.
```

Comment:
I am resolving it for now, but I'll get back to this point.

Reply:
Sorry, missed this comment. Yes, technically the license is part of the code. This was decided long time ago, not sure we thought about it deeply enough.
Not sure how it's different from me scraping a website that uses MUI X pro and extracting the license key that way.

Reply:
The decision to include MUI X Pro in Toolpad was made when we were providing a DX similar to Retool. I believe the upside was to not place Toolpad at a disadvantage for choosing Material UI and not Ant Design for the components. But I can't find back the rationale for it.

I like the idea to require a pro plan to be able to use the pro component. Since we moved a lot closer to pro-code, this decision doesn't seem to make sense anymore. I would remove "MUI X Pro (Data Grid and Date Picker) components are also available for free in Toolpad" from the blog post.

However, I do think that it makes sense for the editor to be able to use MUI X Pro, only not for the final output.

From a legal standpoint, developers have a non-open-source dependency when they look at their tree: https://www.npmjs.com/package/@mui/toolpad-app?activeTab=dependencies. This could work for the editor, we could mention how the Toolpad editor comes with a deployment key of MUI X. For use with other applications, we would either not allow it, or we would need to issue a special license.

Reply:
Removed the statement: https://github.com/mui/material-ui/pull/38015 (PR needs approval)

Reply:
I'm tracking this in https://github.com/mui/mui-toolpad/issues/2325. We can further refine during our weekly call.